### DISCUSSION TITLE
invalid `pydantic.BaseModel` since upgrading to pydantic 2

### DISCUSSION THREAD

**Discussion by djahmag** [0b01001001/spectree#330] [Q&A] [answered]
I've just upgraded to pydantic 2 and seem to have some troubles using the validate annotation.

```
    @spectree_api.validate(
        query=GetTransactionsQueryParams,
        resp=Response(HTTP_200=TransactionsDTO)
    )
    def on_get(self, req: falcon.Request, resp: falcon.Response) -> None:
        """get transactions"""
        ...
```
The error message I get is this:
```
...
../src/interface/api/get/get_transactions.py:28: in GetTransactionsResource
    resp=Response(HTTP_200=TransactionsDTO)
../venv/lib/python3.10/site-packages/spectree/response.py:76: in __init__
    assert issubclass(model, BaseModel), "invalid `pydantic.BaseModel`"
E   AssertionError: invalid `pydantic.BaseModel`
```

I've tried debugging at line 76 in `response.py`: 
`assert issubclass(model, BaseModel), "invalid ``pydantic.BaseModel``"` 
and did find something suspicious:.

When calling `model.__mro__` to check the super classes of my class, I  get a tuple including this class:
`<class 'pydantic.main.BaseModel'>`

When calling `BaseModel.__mro__`, I get this as a result:
`(<class 'pydantic.v1.main.BaseModel'>, <class 'pydantic.v1.utils.Representation'>, <class 'object'>)`
So that seems to use the wrong BaseModel.

Am I using an outdated or wrong method? Or simply the wrong version of something?

I am using SpecTree 1.2.0, pydantic 2.1.1, pydantic-core 2.4.0, falcon 3.1.1 and falcon-swagger-ui 1.2.2

**Comment by kemingy** [comment]
ref to https://github.com/0b01001001/spectree/issues/328#issuecomment-1653581568

**Reply by kemingy** [reply, marked as answer]
I have updated the README at https://github.com/0b01001001/spectree/pull/331.

You can track the progress at https://github.com/0b01001001/spectree/issues/329.

**Reply by djahmag** [reply]
Thanks for the quick response

**Comment by kemingy** [comment]
This is fixed in v1.4.1, you can use pydantic v2 now.
